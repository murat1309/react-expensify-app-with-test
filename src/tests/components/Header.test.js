import React from 'react';
import { shallow } from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  /*  const renderer = new ReactShallowRenderer();
    renderer.render(<Header />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();*/

    //çalıştırdığımızda __snapshots__ adında bir dosyaa oluşturuyor ve onla match'liyor.
    //sen header component'inde herhangi bir değişiklik yaparsan match yapamayacağından test scriptinde 1 fail olacak.
    //ama yaptığın değişikliğin kalılı olmasını istiyorsan konsolu okursan sana 'u' ya basmsanı söyller basarsan component'de yapılan değişikliği günceller ve match'ler.

    //ENZYME Yükledikten sonra
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    //expect(wrapper.find('h1').text()).toBe('Expensify');
});
