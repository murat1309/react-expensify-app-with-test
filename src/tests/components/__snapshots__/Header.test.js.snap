// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header-content"
    >
      <Link
        activeClassName="is-active"
        className="header__title"
        exact={true}
        to="/dashboard"
      >
        <h1>
          Expensify
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
         Logout 
      </button>
    </div>
  </div>
</header>
`;
